jdaviz/core/template_mixin.py

```diff
@@ -1942,6 +1946,244 @@ def __init__(self, *args, **kwargs):
+        # ApertureSubsetSelect instances per-plugin
+        all_aperture_marks = []
+        for viewer in self.image_viewers:
+            # search for existing mark
+            matches = [mark for mark in viewer.figure.marks
+                       if isinstance(mark, ApertureMark)]
+            if len(matches):
+                all_aperture_marks += matches
+                continue
+
+            x_coords, y_coords = self._get_mark_coords(viewer)
+
+            mark = ApertureMark(
+                viewer,
+                x=x_coords,
+                y=y_coords,
+                colors=['#c75109'],
+                fill_opacities=[0.0],
+                visible=self.plugin.is_active)
+            all_aperture_marks.append(mark)
+            viewer.figure.marks = viewer.figure.marks + [mark]
+        return all_aperture_marks
+
+    def _get_mark_coords(self, viewer):
+        if not len(self.selected) or not len(self.dataset.selected):
+            return [], []
+        if self.selected in self._manual_options:
+            return [], []
+
+        if getattr(self, 'multiselect', False):
+            # assume first dataset (for retrieving the region object)
+            # but iterate over all subsets
+            spatial_regions = [self._get_spatial_region(dataset=self.dataset.selected[0], subset=subset)  # noqa
+                               for subset in self.selected if subset != self._manual_options]
+        else:
+            # use cached version
+            spatial_regions = [self.selected_spatial_region]
+
+        x_coords, y_coords = np.array([]), np.array([])
+        for spatial_region in spatial_regions:
+            if spatial_region is None:
+                continue
+
+            if isinstance(spatial_region, PixelRegion):
+                pixel_region = spatial_region
+            else:
+                wcs = getattr(viewer.state.reference_data, 'coords', None)
+                if wcs is None:
+                    return [], []
+                pixel_region = spatial_region.to_pixel(wcs)
+            roi = regions2roi(pixel_region)
+
+            # NOTE: this assumes that we'll apply the same radius factor to all subsets (all will
+            # be defined at the same slice for cones in cubes)
+            if hasattr(roi, 'radius'):
+                roi.radius *= self.scale_factor
+            elif hasattr(roi, 'radius_x'):
+                roi.radius_x *= self.scale_factor
+                roi.radius_y *= self.scale_factor
+            elif hasattr(roi, 'center'):
+                center = roi.center()
+                width_orig = roi.xmax - roi.xmin
+                height_orig = roi.ymax - roi.ymin
+                roi.xmin = center[0] - width_orig/2 * self.scale_factor
+                roi.xmax = center[0] + width_orig/2 * self.scale_factor
+                roi.ymin = center[1] - height_orig/2 * self.scale_factor
+                roi.ymax = center[1] + height_orig/2 * self.scale_factor
```

Comment:
No need to repeat the computation multiple times.
```suggestion
                half_width = abs(roi.xmax - roi.xmin) * 0.5 * self.scale_factor
                half_height = abs(roi.ymax - roi.ymin) * 0.5 * self.scale_factor
                roi.xmin = center[0] - half_width
                roi.xmax = center[0] + half_width
                roi.ymin = center[1] - half_height
                roi.ymax = center[1] + half_height
```

Reply:
can the max ever be larger than the min (is the abs necessary)?